tests/tests_unit/test_data_classes/test_capabilities.py

```diff
@@ -247,6 +250,38 @@ def test_load__action_does_not_exist(self, acl_cls_name: str, bad_action: str, d
 
         assert Capability.load(dumped, allow_unknown=True)
 
+    @pytest.mark.parametrize(
+        "has_write_allscope, has_write_props_allscope, has_write_in_same_scope",
+        itertools.product(*itertools.repeat([True, False], 3)),
+    )
+    def test_load_data_model_instances__with_write_properties(
```

Comment:
Some comments explaining what's going on/being tested here would be good, been staring at it for 4 minutes and I still don't really think I get it 😬